drivers/sensor/MLX90640.zig

```diff
@@ -0,0 +1,910 @@
+
+        for (0..rows) |i| {
+            for (0..columns) |j| {
+                const p = 32 * i + j;
+                const split = 2 * (p / 32 - (p / 64) * 2) + p % 2;
+                self.scratch_data[p] = @floatFromInt((self.eeprom[64 + p] & 0x000E) >> 1);
+                if (self.scratch_data[p] > 3) {
+                    self.scratch_data[p] = self.scratch_data[p] - 8;
+                }
+                const x: f32 = @floatFromInt(@as(u8, 1) << ktaScale2);
+                self.scratch_data[p] = self.scratch_data[p] * x;
+                const y: f32 = @floatFromInt(ktaRC[split]);
+                self.scratch_data[p] = y + self.scratch_data[p];
+                self.scratch_data[p] = self.scratch_data[p] / std.math.pow(f32, 2, @floatFromInt(ktaScale1));
+                //self.scratch_data[p] = self.scratch_data[p] * self.params.offset[p];
+            }
+        }
+
+        var temp: f32 = @abs(self.scratch_data[0]);
+        for (1..pixel_count) |i| {
+            if (@abs(self.scratch_data[i]) > temp) {
+                temp = @abs(self.scratch_data[i]);
+            }
+        }
+
+        ktaScale1 = 0;
+        while (temp < 64) {
+            temp = temp * 2;
+            ktaScale1 = ktaScale1 + 1;
+        }
+
+        for (0..pixel_count) |i| {
+            temp = self.scratch_data[i] * std.math.pow(f32, 2, @floatFromInt(ktaScale1));
+            if (temp < 0) {
+                self.params.kta[i] = @intFromFloat(temp - 0.5);
+            } else {
+                self.params.kta[i] = @intFromFloat(temp + 0.5);
+            }
+        }
+
+        self.params.ktaScale = ktaScale1;
+    }
+
+    fn extract_ky_pixel(self: *Self) void {
+        var KvT: [4]u8 = undefined;
+        var KvRoCo: i8 = @intCast((self.eeprom[52] & 0xF000) >> 12);
+        if (KvRoCo > 7) {
+            KvRoCo = KvRoCo - 16;
+        }
+        KvT[0] = @intCast(KvRoCo);
+
+        var KvReCo: i8 = @intCast((self.eeprom[52] & 0x0F00) >> 8);
+        if (KvReCo > 7) {
+            KvReCo = KvReCo - 16;
+        }
+        KvT[2] = @intCast(KvReCo);
+
+        var KvRoCe: i8 = @intCast((self.eeprom[52] & 0x00F0) >> 4);
+        if (KvRoCe > 7) {
+            KvRoCe = KvRoCe - 16;
+        }
+        KvT[1] = @intCast(KvRoCe);
+
+        var KvReCe: i8 = @intCast(self.eeprom[52] & 0x000F);
+        if (KvReCe > 7) {
+            KvReCe = KvReCe - 16;
+        }
+        KvT[3] = @intCast(KvReCe);
+
+        var kvScale: u8 = @intCast((self.eeprom[56] & 0x0F00) >> 8);
+
+        for (0..rows) |i| {
+            for (0..columns) |j| {
+                const p = 32 * i + j;
+                const split = 2 * (p / 32 - (p / 64) * 2) + p % 2;
+                self.scratch_data[p] = @floatFromInt(KvT[split]);
+                self.scratch_data[p] = self.scratch_data[p] / std.math.pow(f32, 2, @floatFromInt(kvScale));
+                //self.scratch_data[p] = self.scratch_data[p] * mlx90640->offset[p];
```

Comment:
delete?

Reply:
I'll delete it.  I think it was a commented out cpp line that I partially converted to zig and left it there to see if I could understand why it was there.